### ISSUE TITLE
Blocked from fixing Waiting translation

### ISSUE THREAD

**Issue by garretthyder** [GlotPress/GlotPress-WP#853]
Hello,

Not sure what's most appropriate here but I'll explain circumstances before potential fixes.
So there was an original 'correct' translation of a string, now the string is Waiting but is incorrect, so I correct the string and attempt to submit which fails with error;
![screen shot 2017-11-14 at 11 21 11 am](https://user-images.githubusercontent.com/8726005/32800255-ae484d26-c92e-11e7-92a7-0bcf933c0168.png)
To get it to submit I had to accept the incorrect Waiting translation and then find it again and fix it.
I guess I was blocked because the correct translation existed but wasn't the active translation.

So.. not sure what's best but;
Option 1 - Don't block translations if they aren't the currently active one... so change logic that's checks all previous translation to only check the currently active one.
Option 2 - When rejecting a Waiting translation revert to previously accepted translation.

I'm thinking Option #1 is better as theres' cases where the previously accepted translation isn't correct either.

Open to discuss.

Cheers

**Comment by ocean90** [comment]
Did you click on "All translations of this original" before submitting your translations?

**Comment by garretthyder** [comment]
Yes, that's where I found there was another instance before the Waiting string that was correct and blocking the translation submission. So I had to approve the incorrect Waiting string in order to correct it.

**Comment by ocean90** [comment]
> there was another instance before the Waiting string that was correct

So why didn't you approve the correct instance?

**Comment by garretthyder** [comment]
That option didn't seem to be available. There Waiting string that was submitted was more recent so was showing for the string. Is the proper workflow to reject the waiting string, go into All strings, find the correct and approve it?

From the GUI it wasn't very evident that there was any other strings than the one showing as Waiting

**Comment by garretthyder** [comment]
Closing for now after discussion with @ocean90 it was suggested I reject the waiting string and it should revert to the previously accepted version. Will reopen if testing doesn't allow this workflow. Cheers

**Comment by garretthyder** [comment]
Alright @ocean90 I've found a reproduction candidate for this issue;
https://translate.wordpress.org/projects/wp/dev/admin/en-ca/default?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=3852&filters%5Btranslation_id%5D=52836067

So as can be seen the waiting translation is incorrect as it's using a different word than the original string. If you Copy from original on both singular and plural and then try to Add Translation you'll get blocked with the Identical translation exists.

If you reject the string, and then try to submit the corrected string you'll still be blocked.
To get by I have to approve the invalid string and then go back to it and correct it.

I guess the issue is that in All translations there is a rejected, a current and a waiting. The current is correct but the waiting is showing in the translation queue. 

So the question is how to proceed, is there a way to delete the waiting string so the current becomes the active one again. Or is there a way to force submit to replace.

I'm not sure I'll let you take a look and we can discuss.

Appreciated

**Comment by ocean90** [comment]
> So the question is how to proceed, is there a way to delete the waiting string so the current becomes the active one again.

It's worth to mention that the current (green) stays "active" until another translation gets approved. So in this case you'd reject (red) the waiting (orange) string which "deletes" the translation.

**Comment by garretthyder** [comment]
Thanks Dominik, I think I got confused last time as when I 'reject' the item changes to red and stays in view which I have filtered to untranslated so had assumed I then had to go translate it. But refreshing the untranslated view has it disappeared.

I think my confusion was that it stayed around, I understand why now, might be a good call to have a notice or something to go along with reject especially if there's a previous 'current' string that it'll revert to.

Appreciate you taking a look
